### ISSUE TITLE
Tileset ordering in tmx file causing exception

### ISSUE THREAD

**Issue by danielgholmes** [megamarc/Tilengine#113]
From the .tmx file, the tilesets are referenced like so
```
 <tileset firstgid="357" source="../tilesets/test.tsx"/>
 <tileset firstgid="362" source="../tilesets/test2.tsx"/>
 <tileset firstgid="367" source="../tilesets/test3.tsx"/> 
....
```

What I noticed is that when new tilesets are added to a tmx file, Tiled appends it to the end of the above list in the file. Then when trying to setup a layer with this new tileset it throws a `'Invalid Tileset reference'` exception. However, if I manually edit the tmx file and put the newly referenced tileset at the top of the tileset list, then it works fine.

Basically, it seems the order in which tilesets are referenced in the tmx file causes issues with Tilengine, particularly when setting up layers.

**Comment by megamarc** [comment]
Hi Daniel,
I can't reproduce the issue. Here I attach a sample layer using three distinct tilesets:
[assets.zip](https://github.com/megamarc/Tilengine/files/10831702/assets.zip)
![layer](https://user-images.githubusercontent.com/14149983/221368503-5c40d86b-8086-4d77-9253-04d3b8cf45eb.png)

Editing the tmx file by hand, shuffling the order of the tilesets doesn't cause any error, the layer is renderer correctly

To debug this issue I need a test asset that reproduces the poblem
